### PR TITLE
acl: JWT as SSO auth method

### PR DESCRIPTION
**Pull request by pkazmierczak**
This PR introduces a new ACL Auth Method: JWT, which allows users to exchange 3rd party JSON Web Tokens for Nomad ACL Tokens. We achieve this by:
- extending ACLAuthMethodConfig fields with JWT-specific properties
- a new RPC endpoint `ACL.Login`
- a new HTTP API endpoint `/v1/acl/login`
- adjustments to the CLI that allow for new `-type=JWT` when creating and updating auth methods, and to the `nomad login` command
- and adjustments to the sign in UI which now detects if there are JWT auth methods present, whether they are multiple and modifies the "sign in with token" field accordingly. 

All the changes mentioned above were described in [NMD-167](https://docs.google.com/document/d/1MqhdDpMqOYMT72PgdhxsGHmF0-WcK5425oxEBAkQRXo/edit), and all the commits in this pull request (except for changelog) were previously reviewed.

### CODE REVIEW

File: lib/auth/jwt/validator.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		return nil, fmt.Errorf("unable to get validation keys from JWKS: %v", err)
+	}
+	return keySet, nil
+}
+
+func usingOIDC(ctx context.Context, oidcurl string, oidccapem []string) (jwt.KeySet, error) {
+	// Measure the OIDC endpoint performance.
+	defer metrics.MeasureSince([]string{"nomad", "acl", "jwt", "oidc_jwt"}, time.Now())
+
+	// TODO why do we have DiscoverCaPem as an array but JWKSCaPem as a single string?
```
Comment (tgross) [inline lib/auth/jwt/validator.go:114]: Just a quick check that this TODO is ok to leave before merging?

File: nomad/acl_endpoint.go [hunk 2]
```diff
@@ -2777,3 +2783,152 @@ func (a *ACL) OIDCCompleteAuth(
...
+	// respect stale queries.
+	if done, err := a.srv.forward(structs.ACLLoginRPCMethod, args, args, reply); done {
+		return err
+	}
+
+	// Measure the login endpoint performance.
+	defer metrics.MeasureSince([]string{"nomad", "acl", "login"}, time.Now())
+
+	// This endpoint can only be used once all servers in all federated regions
+	// have been upgraded to 1.5.2 or greater, since JWT Auth method was
```
Comment (tgross) [inline nomad/acl_endpoint.go:2806]: Should this read 1.5.3? Or should we just remove the specific version here and let that live in the `minACLJWTAuthMethodVersion` constant so we don't need to update it if we have to ship a security update or whatever before this goes out?

**Review by tgross — APPROVED** [review]
LGTM!

File: lib/auth/jwt/validator.go (hunk 1 above)
Comment (pkazmierczak) [reply lib/auth/jwt/validator.go:114]: Right, this is "a note for future us." Something we'll definitely revisit.

**Pull request merged**